### DISCUSSION TITLE
Sonoff ZbBridge & Devices IDs mutation

### DISCUSSION THREAD

**Discussion by alsak0de** [arendst/Tasmota#21034] [Q&A]
I have around 40 ZB devices connected to a tasmotised sonoff zb bridge. My problem is that the messages I see in the tasmota console or the mqtt publications from the zbbridge include only a short 4-digit HEX ID instead of the full IEEE address/id. This short ID keeps mutating for a big portion of the devices making them hard to trace and operate upon. Is there anyway to ask the tasmota bridge to push long indentifiers/adresses instead of these short HEX IDs?

**Comment by sfromis** [comment]
You have a basic problem in your setup if the "short address" keeps mutating.

Anyway, just do not use that "short address" for anything, recommended is to use the `ZbName` command to assign an alias to the full IEEE address.

**Comment by alsak0de** [comment]
Thank you but. I use ZbName but to link the alias to the short address not the IEEE so it ends up with the same problem.
How do I collect the IEEE addresses? I can only see a few of them showing in the data published in console. 
Any way I can access the Sonoff ZbBridge via API to gather more data?

**Reply by sfromis** [reply]
I specifically recommended to use `ZbName` with the IEEE address, **not** the short address, which I recommended to *never* use. `IEEEAddr` is indeed included in the output from `ZbInfo`, but when you see zeros there, something is rotten.

**Comment by alsak0de** [comment]
Trying to get IEEE with ZbInfo:
17:56:44.776 CMD: zbinfo 0xA546
17:56:44.784 MQT: tele/SonoffZBGW/A546/SENSOR = {"ZbInfo":{"0xA546":{"Device":"0xA546","IEEEAddr":"0x0000000000000000","Endpoints":[],"Config":[],"Reachable":true,"LastSeen":169,"LastSeenEpoch":1711558435,"LinkQuality":47}}}

**Reply by s-hadinger** [reply]
If IEEE address is zero, try pairing again. Sometimes devices fail to announce their address

**Comment by alsak0de** [comment]
Thank you. I figured out a solution, the 0x0000000 is shown with ZbInfo but if I do a ZbPing I obtain the right address.
I've created now a script to programatically run the ZbPings for every device that shows up in zbinfo responses and store the IEEE address to then run a ZbName against the long address instead of short.

Thanks for the hints!
